crates/ruma-client-api/src/relations/get_events_relating_to_event.rs

```diff
@@ -0,0 +1,279 @@
+
+            let mut url = format!(
+                "{}{}",
+                base_url.strip_suffix('/').unwrap_or(base_url),
+                ruma_common::api::select_path(
+                    considering_versions,
+                    &METADATA,
+                    Some(format_args!(
+                        "/_matrix/client/unstable/rooms/{}/relations/{}",
+                        room_id_percent, event_id_percent
+                    )),
+                    None,
+                    None,
+                )?
+            );
+
+            if let Some(rel_type) = self.rel_type {
+                url.push('/');
+                url.push_str(&Cow::from(utf8_percent_encode(rel_type.as_str(), NON_ALPHANUMERIC)));
+
+                if let Some(event_type) = self.event_type {
+                    url.push('/');
+                    url.push_str(&Cow::from(utf8_percent_encode(
+                        &event_type.to_string(),
+                        NON_ALPHANUMERIC,
+                    )));
+                }
+            }
+
+            #[derive(Debug, serde::Serialize)]
+            struct RequestQuery<'a> {
+                from: Option<&'a str>,
+                to: Option<&'a str>,
+                limit: Option<UInt>,
+            }
+
+            let query = RequestQuery { from: self.from, to: self.to, limit: self.limit };
+            let query_string = ruma_common::serde::urlencoded::to_string(query)?;
+
+            if !query_string.is_empty() {
+                url.push('?');
+                url.push_str(&query_string);
+            }
+
+            http::Request::builder()
+                .method(http::Method::GET)
+                .uri(url)
+                .header(header::CONTENT_TYPE, "application/json")
+                .header(
+                    header::AUTHORIZATION,
+                    format!(
+                        "Bearer {}",
+                        access_token
+                            .get_required_for_endpoint()
+                            .ok_or(ruma_common::api::error::IntoHttpError::NeedsAuthentication)?,
+                    ),
+                )
+                .body(T::default())
+                .map_err(Into::into)
+        }
+    }
+
+    #[cfg(feature = "server")]
+    impl ruma_common::api::IncomingRequest for IncomingRequest {
+        type EndpointError = crate::Error;
+        type OutgoingResponse = Response;
+
+        const METADATA: ruma_common::api::Metadata = METADATA;
+
+        fn try_from_http_request<B, S>(
+            request: http::Request<B>,
+            path_args: &[S],
+        ) -> Result<Self, ruma_common::api::error::FromHttpRequestError>
+        where
+            B: AsRef<[u8]>,
+            S: AsRef<str>,
+        {
+            use ruma_common::{OwnedEventId, OwnedRoomId};
+
+            // FIXME: find a way to make this if-else collapse with serde recognizing trailing
+            // Option
```

Comment:
Is this comment still relevant? Or sub `if-else` with `match`, maybe?

Reply:
Oh right, that's a copy-pasted comment from `state::get_state_events_for_key`'s implementation. I don't know if it's still relevant though.

Reply:
Actually this comment is probably referring to #1089, so I should probably just replace `if-else`.